### PR TITLE
[FrameworkBundle][TwigBundle] Make EngineInterface autowirable

### PR DESCRIPTION
**Pull request by dunglas**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

**Comment by Ener-Getick** [comment]
Isn't this option supposed to be defined by the user to choose which service to inject in priority ?
Are there multiple services corresponding to this interfaces ?

**Comment by xabbuh** [comment]
:+1:

**Comment by Ener-Getick** [comment]
I answered the question myself :+1:

**Comment by fabpot** [comment]
Thank you @dunglas.

File: src/Symfony/Bundle/TwigBundle/Resources/config/twig.xml [hunk 1]
```diff
@@ -59,6 +59,9 @@
             <argument type="service" id="twig" />
             <argument type="service" id="templating.name_parser" />
             <argument type="service" id="templating.locator" />
+
+            <autowiring-type>Symfony\Component\Templating\EngineInterface</autowiring-type>
+            <autowiring-type>Symfony\Bundle\FrameworkBundle\Templating\EngineInterface</autowiring-type>
```
Comment (stof) [inline src/Symfony/Bundle/TwigBundle/Resources/config/twig.xml]: Wouldn't this lead to ambiguous types when loading both engines ?

Comment (dunglas) [inline src/Symfony/Bundle/TwigBundle/Resources/config/twig.xml]: No, the last defined take precedence. If both are loaded, twig will be autowired (it it is registered after FrameworkBundle).

Comment (stof) [inline src/Symfony/Bundle/TwigBundle/Resources/config/twig.xml]: if both engines are enabled, the delegating engine should be used, as this is what the ``templating`` service will be.

Btw, if you configure only the PHP engine, your PR is also broken as the delegating engine won't be used either for the ``templating`` service.

This config should be set in the compiler pass defining what the ``templating`` service is

**Pull request merged**